### PR TITLE
Fix for GaussSplitter with GaudiExec

### LINKED ISSUE #840: GaussSplitter not overriding LHCbApp for GaudiExec
I paste the email below. We had a problem before with the GaussSplitter overriding the LHCbApp options but it seems here to still not interacting with them properly.

I have 2 different jobs with 2 subjobs each. The former uses a private version of Gauss with GaudiExec, the latter a default Gauss version. Both use GaussSplitter

 GaussSplitter (
   numberOfJobs = 2,
   eventsPerJob = 50,
   firstEventNumber = 0
 )

as splitter. In the option file I setup the number of events to generate with

LHCbApp().EvtMax = 100

If I run the default version of Gauss then I get 50 events per subjobs correctly uncorrelated --> Don't know why the NextNext event (spillover event) seems to have the same seeds between subjobs, but the main event have different seeds so it is fine
If I run the same option file with GaudiExec then I get 100 events per subjobs, 50 of each are totally correlated among the 2 subjobs since they have the same seeds.
I know I can easily avoid the problem if I change

LHCbApp().EvtMax = 100 --> LHCbApp().EvtMax = 50

when I run with GaudiExec, but I'm wondering if this is a bug or a known behaviour.

### PR DESCRIPTION
**Pull request by mesmith75**
Fixes #840. Quick fix for the number of events in GaussSplitter for GaudiExec.

### CODE REVIEW

File: python/GangaLHCb/Lib/Splitters/GaussSplitter.py [hunk 1]
```diff
@@ -77,7 +77,11 @@ def split(self, job):
             first = self.firstEventNumber + i * self.eventsPerJob + 1
             opts = 'from Gaudi.Configuration import * \n'
             opts += 'from Configurables import GenInit \n'
-            opts += 'ApplicationMgr().EvtMax = %d\n' % self.eventsPerJob
+            if isinstance(job.application, GaudiExec):
```
Comment (mesmith75) [inline python/GangaLHCb/Lib/Splitters/GaussSplitter.py]: @egede Unfortunately not. It seems as if for the old code the options are pickled, so subsequent changes to LHCbApp undo any options for it in the options files (eg. ```LHCbApp().DDDBtag```). For GaudiExec though, the ```LHCbApp().EvtMax``` takes precedence over ```ApplicationMgr().EvtMax``` so we need to use LHCbApp for the GaudiExec.

Ok, I can switch it around to test for Gauss application instead.

**Comment by mesmith75** [comment]
If there are no objections I will merge this and start the release for 6.2.3

File: python/GangaLHCb/Lib/Splitters/GaussSplitter.py [hunk 2]
```diff
@@ -77,7 +77,11 @@ def split(self, job):
             first = self.firstEventNumber + i * self.eventsPerJob + 1
             opts = 'from Gaudi.Configuration import * \n'
```
Comment (rob-c) [inline python/GangaLHCb/Lib/Splitters/GaussSplitter.py]: Is this line still needed? The `GenInit` is used but I don't see anything other than the `ApplicationMgr` which may be imported here, or am I mis-reading the code?

Comment (mesmith75) [inline python/GangaLHCb/Lib/Splitters/GaussSplitter.py]: Yes you need to ```import ApplicationMgr``` - I guess this can be moved to the relevant part.

Comment (mesmith75) [inline python/GangaLHCb/Lib/Splitters/GaussSplitter.py]: It is needed to import ```ApplicationMgr``` - I guess I could move this to be specifically under the Gauss application.

**Pull request merged**